src/main/groovy/com/bmuschko/gradle/docker/utils/DockerThreadContextClassLoader.groovy

```diff
@@ -499,4 +516,57 @@ class DockerThreadContextClassLoader implements ThreadContextClassLoader {
     private Class loadBindClass() {
         loadClass("${MODEL_PACKAGE}.Bind")
     }
+    
+    private def parseDevice(String deviceString) {
+        List<String> tokens = deviceString.tokenize(':')
+        int numberOfTokens = tokens.size()
+
+        String permissions = 'rwm'
+        String source = ''
+        String destination = ''
+        
+        switch(numberOfTokens) {
+            case 3:
+                if (validDeviceMode(tokens[2])) {
+                    permissions = tokens[2]
+                } else {
+                    throw new IllegalArgumentException("Invalid device specification: " + deviceString)
+                }
+            case 2:
+                if (validDeviceMode(tokens[1])) {
+                    permissions = tokens[1]
+                } else {
+                    destination = tokens[1]
+                }
+            case 1:
+                source = tokens[0]
+                break
+            default:
+                throw new IllegalArgumentException("Invalid device specification: " + deviceString)
+        }
+        
+        if (!destination) {
+            destination = source
+        }
+        
+        return [permissions, destination, source]
+    }
+    
+    private boolean validDeviceMode(String deviceMode) {
+        Map<String, Boolean> validModes = [r: true, w: true, m: true]
+
+        if (!deviceMode) {
+            return false
+        }
+
+        for (char ch in deviceMode) {
```

Comment:
Can we do this in a more terse/groovy way?

	deviceMode.each {
		if (!validModes.get(it)) { return false
		} else { validModes.put(it, false) }
	}